fix(openrpc): add initial support for key management / signing APIs

shanejonas opened the pull request:
you can preview this document here: https://playground.open-rpc.org/?schemaUrl=https://raw.githubusercontent.com/etclabscore/jade-signer-rpc/fix/openrpc-document/signer/openrpc.json

shanejonas committed:
fix(openrpc): add initial support for key management / signing APIs

shanejonas commented:
Should we keep the method name prefix `jade_`?

I'm thinking `signer_`

BelfordZ commented on signer/openrpc.json:
```diff
@@ -21,18 +23,18 @@
         "description": "Seconds since Jan 01 1970 (UTC)",
         "schema": {
-          "type": "integer",
-          "format": "int64"
```
why remove the int64? 

If anything, dropping it to int16 would be better.

shanejonas replied:
because its not a format JSON Schema supports. number is just any number, no bounds, its up to the language.

theres a big rabbit hole if you want to go down it:
https://github.com/json-schema-org/json-schema-spec/issues/574

shanejonas replied:
or you can do `minimum: 0` and `maximum: ` if we want to go that route

BelfordZ commented on signer/openrpc.json:
```diff
@@ -145,43 +97,158 @@
       "result": {
-        "name": "signTransactionResult",
-        "description": "",
+        "name": "signedTransaction",
```
The above methods result is `newAccountResult`, how come we are changing that pattern for this one?

shanejonas replied:
sounded better lol 😂 thoughts?

BelfordZ replied:
I dont have any pref, just as long as its consistent

shanejonas commented:
would be nice to figure out a more generic approach to `{chain: 'etc'}`, or atleast `enum` through all the chains we support in `multi-geth`?

shanejonas commented on signer/openrpc.json:
```diff
@@ -145,43 +97,158 @@
         "name": "newAccountResult",
         "schema": {
-          "type": "boolean"
+          "$ref": "#/components/contentDescriptors/Address"
```
this is wrong. cant `$ref` a content descriptor inside a `schema: {}`. oops.

BelfordZ replied:
wtb tests :)

BelfordZ replied:
https://github.com/etclabscore/jade-signer-rpc/issues/19

BelfordZ commented:
@shanejonas yeah,. Agreed. 

I was thinking about this... what is different about each chain? at a specific block number, the block hash will be different. That being said, Maybe we could try something like

`{ chain: { blockNumber: "0x123", blockHash: "0x1231231" } }`

shanejonas committed:
fix(openrpc): rename jade_ to signer_ and fix schema issues

shanejonas commented:
@r8d8 @BelfordZ renamed all methods to use the `signer_` prefix.

I've also included `signer_sign` but it is **not** yet implemented. I've created #15 to address that.

and I've also changed the way the passphrase is taken for the existing `signTransaction` to be more consistent with `signer_sign` and the rest of the `eth` APIs. I've created #22 to address that.

This also deleted **clientVersion** and **heartbeat** since we will have `rpc.discover` available. I've created #23 and #24 to address that.

shanejonas committed:
fix: add mnemonic methods
